sklearn/discriminant_analysis.py

```diff
@@ -462,7 +464,7 @@ def _solve_svd(self, X, y):
         # 2) Within variance scaling
         X = np.sqrt(fac) * (Xc / std)
         # SVD of centered (within)scaled data
-        U, S, Vt = linalg.svd(X, full_matrices=False)
+        U, S, Vt = linalg.svd(X, full_matrices=False, check_finite=False)
```

Comment:
if std is small the input could be non finite
(similar to https://github.com/scikit-learn/scikit-learn/pull/18850/files#r526344155 )

Reply:
This should be safe because we will never have a division by zero